disnake/embeds.py

```diff
@@ -794,3 +797,66 @@ def _handle_resource(self, url: Optional[Any], file: File, *, key: _FileKey) ->
         else:
             self._files.pop(key, None)
             return str(url) if url is not None else None
+
+    def check_limits(self):
+        """
+        Checks if this embed fits within the limits dictated by Discord.
+        There is also a 6000 character limit across all embeds in a message.
+
+        +--------------------------+------------------------------------+
+        |   Field                  |              Limit                 |
+        +--------------------------+------------------------------------+
+        | title                    |        256 characters              |
+        +--------------------------+------------------------------------+
+        | description              |        4096 characters             |
+        +--------------------------+------------------------------------+
+        | fields                   |        Up to 25 field objects      |
+        +--------------------------+------------------------------------+
+        | field.name               |        256 characters              |
+        +--------------------------+------------------------------------+
+        | field.value              |        1024 characters             |
+        +--------------------------+------------------------------------+
+        | footer.text              |        2048 characters             |
+        +--------------------------+------------------------------------+
+        | author.name              |        256 characters              |
+        +--------------------------+------------------------------------+
+
+        .. versionadded:: 2.6
+
+        Returns:
+            bool: True if the embed fits within Discord limits
+
+        Raises:
+            ValueError: One or more of the embed attributes are too long.
+        """
+
+        if self.title and len(self.title.strip()) > 256:
+            raise ValueError("Embed title cannot be longer than 256 characters")
+
+        if self.description and len(self.description.strip()) > 4096:
+            raise ValueError("Embed description cannot be longer than 4096 characters")
+
+        if self._footer and len(self._footer["text"].strip()) > 2048:
```

Comment:
```suggestion
        if self._footer and len(self._footer.get("text", "").strip()) > 2048:
```
While the `text` field *should* always exist according to the docs, those have been wrong before (particularly the embed docs) so I'd rather be careful here :sweat_smile: